### PR TITLE
only build crates for default target unless asked

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
fixes https://github.com/rust-lang/docs.rs/issues/343

This PR makes it so that we don't automatically build crates for every target in our build container. See discussion in the linked issue for rationale.

I've also included an expansion of the documentation of the package metadata, since it wasn't very well explained before.

### CODE REVIEW

**Comment by nagisa** [comment]
*How* to enable alternate targets should be made very visible. Not squirrelled somewhere in the annals of documentation, but as a link on the pop-out where the list of targets was originally. FWIW majority of my crates are cross-platform and expose platform-specific modules and this is the first time I hear about the docs.rs specific cargo.toml section.

File: templates/about.hbs [hunk 1]
```diff
@@ -101,21 +101,44 @@
...
+# Features to pass to Cargo (default: none)
 features = [ "feature1", "feature2" ]
+
+# Whether to pass `--all-features` to Cargo (default: false)
 all-features = true
+
+# Whether to pass `--no-default-features` to Cargo (default: false)
 no-default-features = true
+
+# Target to test build on, used as the default landing page (default: "x86_64-unknown-linux-gnu")
```
Comment (dekellum) [inline templates/about.hbs]: Thanks, this now looks very clear to me. Very helpful.

**Comment by dekellum** [comment]
Regarding the docs discovery problem,  perhaps the [cargo reference, manifest format][https://doc.rust-lang.org/cargo/reference/manifest.html] could a special note and link to https://docs.rs/about#metadata-for-custom-builds (ideally with that anchor id working)?

**Comment by jyn514** [comment]
What is this waiting on? I would like to see this to reduce build times.

**Comment by pietroalbini** [comment]
Well, for sure this PR will have to be rewritten from scratch, as `chroot_builder` (thankfully) doesn't exist anymore.

**Comment by jyn514** [comment]
Closing in favor of https://github.com/rust-lang/docs.rs/pull/532 (as mentioned above, this was obsoleted by Rustwide)

**Pull request closed without merging**